File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    if (nums === null || nums.length === 0) return 0;

    let maxSeq = 0;
    const numsSet = new Set(nums);

    for (let i = 0; i < nums.length; i++) {
        if (numsSet.has(nums[i] - 1)) {
            continue;
        }

        let currNum = nums[i];
        let currMax = 1;

        while (numsSet.has(currNum + 1)) {
            currNum++;
            currMax++;
        } 
        maxSeq = Math.max(maxSeq, currMax);
    }

    return maxSeq;
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        longest_seq = 0
        
        
        for i in range(0, len(nums)):
            num = nums[i]
            prev = num - 1
            
            next_num = num + 1
            current_seq = 1
           
            if prev not in nums_set:
                while next_num in nums_set:
                    next_num += 1
                    current_seq += 1
            
                longest_seq = max(longest_seq, current_seq)
        
        
        return longest_seq